### PR TITLE
fix(Summary Page): displays all answered questions

### PR DESCRIPTION
**Pull request by froehlichcr**
#60

- addresses bug in reducer logic after data model shift to arrays

### CODE REVIEW

File: src/components/lib/Questions.tsx [hunk 1]
```diff
@@ -22,12 +23,19 @@ export abstract class Questions {
...
+    // TODO: circle back and fix this logic. The problem is that our reducer is merging by KEY,
+    // which in the case of arrays is the index, and the index will always be 0 if we're passing in new arrays
+    // There are cleaner ways to do this.
+    props.form.responses = props.form.responses || [];
+    const value          = props.form.responses.find((r) => r.id === props.step.id);
+    if (!value) {
+      props.form.responses.push(props.step);
+    } else {
+      merge(value, props.step);
+    }
```
Comment (froehlichcr) [inline src/components/lib/Questions.tsx:35]: This is an abomination, and I cannot wait to rip it out.

File: src/components/lib/Pages.tsx [hunk 2]
```diff
@@ -46,17 +53,19 @@ export abstract class Pages {
...
+    return questionnaire.getResults(props.form).map((result) => (
       <li key={`${props.stepId}_${result.id}`} className="padding-bottom-2">
         <span>
           {result.label}:{'  '}
           <b>{result.name}</b>
         </span>
         <div
           className="text-light"
-          dangerouslySetInnerHTML={{ __html: Pages.getReason(props, result, global) }}
+          dangerouslySetInnerHTML={{
```
Comment (TomNUSDS) [inline src/components/lib/Pages.tsx:64]: This is interesting to think through. Can someone in the org mistakenly type in some html tags for the reason generate bad html? Something to follow up with in testing.

File: src/composable/Questionnaire.ts [hunk 3]
```diff
@@ -55,7 +55,10 @@ export class Questionnaire implements IQuestionnaire {
     Object.assign(this, data);
 
     // Create a new collection for our flow logic
-    this.steps = [...this.questions];
+    this.steps = this.questions.map((q, i) => ({
```
Comment (TomNUSDS) [inline src/composable/Questionnaire.ts:58]: Maybe a longer comment? It seems like this just adds an `order` to the elements to preserve the order in the array?

Comment (TomNUSDS) [reply src/composable/Questionnaire.ts:58]: Ahh, I see the comment below where `order` is declared! Maybe something similar here?

**Review by TomNUSDS — APPROVED** [review]
LGTM

File: src/components/lib/Pages.tsx (hunk 2 above)
Comment (froehlichcr) [reply src/components/lib/Pages.tsx:64]: It's a potential risk. The trick is to allow very specific customizations within labels, and this is the way to do that generically.

File: src/composable/Questionnaire.ts (hunk 3 above)
Comment (froehlichcr) [reply src/composable/Questionnaire.ts:58]: Technically, this is no longer required--just a remnant from one of many attempts to fix this. Doesn't hurt anything now; will circle back and gut it later.

**Pull request merged**